### PR TITLE
Add v1.2 Features for Onboarding Guide 

### LINKED ISSUE #100: Add Feature - Onboarding guide prompt

### LINKED ISSUE #104: Add feature - Onboarding guide instructions

### LINKED ISSUE #106: Add feature - Onboarding guide: Highlight area

### LINKED ISSUE #109: Add feature - Onboarding Guide (Add person)

### LINKED ISSUE #111: Add feature - Onboarding (View persons)

### PR DESCRIPTION
**Pull request by Wei-Xinn**
This PR adds the v1.2 features for the onboarding guide

## TODO
- [x] Prompt ( Onboarding is currently accessed via menu item) 

## Closes
- closes #104 
- closes #106 
- closes #109 
- closes #111
- closes #100

### CODE REVIEW

File: src/main/java/seedu/contax/MainApp.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
     protected Storage storage;
     protected Model model;
     protected Config config;
+    private boolean isFirstRun;
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/MainApp.java:48]: This flag is not used as far as I can tell.
```suggestion

```

File: src/main/java/seedu/contax/model/OnboardingStory.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package seedu.contax.model;
+
+import java.util.ArrayList;
+
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/model/OnboardingStory.java:4]: Likewise, I feel the overall class should at least have a javadoc header.

**Review by sharpstorm — CHANGES_REQUESTED** [review]
Generally small issues, but I think some test cases would be great.

File: src/main/java/seedu/contax/ui/UiManager.java [hunk 3]
```diff
@@ -44,6 +44,10 @@ public void start(Stage primaryStage) {
             mainWindow.show(); //This should be called before creating other UI parts
             mainWindow.fillInnerParts();
 
+            if(isFirstRun) {
```
Comment (HanJiyao) [inline src/main/java/seedu/contax/ui/UiManager.java:47]: same here

File: src/main/java/seedu/contax/MainApp.java [hunk 4]
```diff
@@ -66,6 +67,10 @@ public void init() throws Exception {
         logic = new LogicManager(model, storage);
 
         ui = new UiManager(logic);
+
+        if(model.getAddressBook().equals(SampleDataUtil.getSampleAddressBook())) {
```
Comment (HanJiyao) [inline src/main/java/seedu/contax/MainApp.java:71]: just a minor checkstyle if shd follow by white space

File: src/main/java/seedu/contax/MainApp.java (hunk 1 above)
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/MainApp.java:48]: FIxed

File: src/main/java/seedu/contax/ui/UiManager.java (hunk 3 above)
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/UiManager.java:47]: Noted fixed

File: src/main/java/seedu/contax/model/OnboardingStep.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package seedu.contax.model;
+
+import seedu.contax.model.person.Person;
+
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/model/OnboardingStep.java:6]: Added

File: src/main/java/seedu/contax/model/OnboardingStory.java (hunk 2 above)
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/model/OnboardingStory.java:5]: Added

File: src/main/java/seedu/contax/model/OnboardingStory.java [hunk 6]
```diff
@@ -20,4 +24,33 @@ public OnboardingStep getStep(int index) {
...
+        if (otherStory.getSize() != getSize()) {
+            return false;
+        }
+
+        for (int i = 0; i < getSize(); i++) {
+            if (!otherStory.getStep(i).equals(getStep(i))) {
+                return false;
+            }
+        }
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/model/OnboardingStory.java:47]: You can simplify this to 
```suggestion
        otherStory.story.equals(story);
```

Comment (Wei-Xinn) [reply src/main/java/seedu/contax/model/OnboardingStory.java:47]: Thanks for the suggestion. Fixed.

**Review by sharpstorm — APPROVED** [review]
LGTM.

**Review by HanJiyao — APPROVED** [review]
LGTM

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm! merging

**Pull request merged**